test/parallel/test-fs-write-stream-autoclose-option.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const path = require('path');
+const fs = require('fs');
+
+var file = path.join(common.tmpDir, 'write-autoclose-opt1.txt');
+common.refreshTmpDir();
+var stream = fs.createWriteStream(file, {flags: 'w+', autoClose: false});
+stream.write('Test1');
+stream.end();
+stream.on('finish', common.mustCall(function() {
+  process.nextTick(common.mustCall(function() {
+    assert.strictEqual(stream.closed, undefined);
+    assert.strictEqual(!stream.fd, false);
+    next();
+  }));
+}));
+
+function next() {
+  // This will tell us if the fd is usable again or not
+  stream = fs.createWriteStream(null, {fd: stream.fd, start:0 });
+  stream.write('Test2');
+  stream.end();
+  stream.on('finish', common.mustCall(function() {
+    assert.strictEqual(stream.closed, true);
+    assert.strictEqual(stream.fd, null);
+    process.nextTick(common.mustCall(function() {
```

Comment:
I don't think you need the anonymous function. I think you can just have `common.mustCall(next2)`.